forc/src/ops/forc_check.rs

```diff
@@ -20,5 +20,10 @@ pub fn check(command: CheckCommand) -> Result<CompileResult<ty::TyProgram>> {
     let manifest = PackageManifestFile::from_dir(&this_dir)?;
     let plan = pkg::BuildPlan::from_lock_and_manifest(&manifest, locked, offline)?;
 
-    Ok(pkg::check(&plan, terse_mode)?.flat_map(|(_, tp)| CompileResult::new(tp, vec![], vec![])))
+    let mut v = pkg::check(&plan, terse_mode)?;
+    let res = v
+        .pop()
```

Comment:
I wonder if there should be an optional argument to `pkg::check` that asserts there's at least one element in the result for you, since we check this in a few places.

Reply:
I think it's fine to keep it like this as we are already returning an error [here](https://github.com/FuelLabs/sway/pull/3212/files#diff-f398539dbba9474754d61669ea72772f8de39c155c10a613069b7f2b0e0158a4R2558) if there are no elements to return.